### PR TITLE
get and show cmdline when earlyomm kill a process

### PR DESCRIPTION
**Pull request by RanHuang**
There are multiple java processes in my system. When earlyoom determines to kill a process, the process name is just 'java'. I need more information about the killed java process. e.g: 
* run earlyoom without get-cmdline
image

* run earlyoom with get-cmdline
image

### CODE REVIEW

**Comment by RanHuang** [comment]
#264

File: MANPAGE.md [hunk 1]
```diff
@@ -161,6 +161,9 @@ processes when earlyoom determines to kill processes.
 #### \-\-dryrun
 dry run (do not kill any processes)
 
+### \-\-get-cmdline
```
Comment (rfjakob) [inline MANPAGE.md:164]: No need for a flag - let's just have it enabled all the time.

File: meminfo.c [hunk 2]
```diff
@@ -250,6 +250,41 @@ int get_comm(int pid, char* out, size_t outlen)
...
+        int fread_errno = errno;
+        perror("get_cmdline: fread() failed");
+        fclose(f);
+        return -fread_errno;
+    }
+    fclose(f);
+    if (n < 1) {
+        return -ENODATA;
+    }
+    /* replace new line charater with space */
```
Comment (rfjakob) [inline meminfo.c:276]: Comment should be: "replace null character with space"

File: meminfo.c [hunk 3]
```diff
@@ -250,6 +250,41 @@ int get_comm(int pid, char* out, size_t outlen)
...
+    for (size_t i = 0; i < n; i++) {
+        if (out[i] == '\0') {
+            out[i] = ' ';
+        }
+    }
+    // Strip trailing newline
```
Comment (rfjakob) [inline meminfo.c:282]: Comment should be: "Strip trailing space"

File: testsuite_unit_test.go [hunk 4]
```diff
@@ -187,6 +187,26 @@ func Test_get_comm(t *testing.T) {
...
+func Test_get_cmdline(t *testing.T) {
+	pid := os.Getpid()
+	res, comm := get_cmdline(pid)
+	if res != 0 {
+		t.Fatalf("error %d", res)
+	}
+	if len(comm) == 0 {
+		t.Fatalf("empty process cmdline %q", comm)
+	}
+	t.Logf("process name %q", comm)
```
Comment (rfjakob) [inline testsuite_unit_test.go:199]: Should be "process cmdline" here

**Review by rfjakob — CHANGES_REQUESTED** [review]
Nice, thank you! A few details to change, please

**Comment by RanHuang** [comment]
I've fixed all the issues

File: kill.c [hunk 5]
```diff
@@ -346,6 +352,11 @@ bool is_larger(const poll_loop_args_t* args, const procinfo_t* victim, procinfo_
             debug("pid %d: error reading process name: %s\n", cur->pid, strerror(-res));
             return false;
         }
+        res = get_cmdline(cur->pid, cur->cmdline, sizeof(cur->cmdline));
+        if (res < 0) {
+            debug("pid %d: error reading process cmdline: %s\n", cur->pid, strerror(-res));
+            return false;
+        }
```
Comment (rfjakob) [inline kill.c:359]: Move after the if block, not inside

File: kill.c [hunk 6]
```diff
@@ -294,6 +295,11 @@ bool is_larger(const poll_loop_args_t* args, const procinfo_t* victim, procinfo_
...
```
Comment (rfjakob) [inline kill.c:302]: We don't need cmdline yet, remove here and only do this once, at the end of the function.

File: testsuite_cli_test.go [hunk 7]
```diff
@@ -63,7 +63,7 @@ func parseMeminfo() (memTotal int64, swapTotal int64) {
 // earlyoom RSS should never be above 1 MiB,
 // but on some systems, it is (due to glibc?).
 // https://github.com/rfjakob/earlyoom/issues/221
-const rssMaxKiB = 2048
+const rssMaxKiB = 4096
```
Comment (rfjakob) [inline testsuite_cli_test.go:66]: This is a different topic, please remove from this pull request

**Review by rfjakob — CHANGES_REQUESTED** [review]

**Comment by rfjakob** [comment]
Merged, thanks!

**Pull request merged**